skorch/tests/test_helper.py

```diff
@@ -966,8 +976,69 @@ def train_step_single(self, *args, **kwargs):
             device=None,
             accelerator=accelerator,
             max_epochs=2,
+            callbacks=[('lr_scheduler', LRScheduler)],
         )
         X, y = data
         # does not raise
         net.fit(X, y)
         net.predict(X)
+
+        # make sure that even after resetting parameters, components are still prepared
+        net.set_params(
+            module__hidden_units=7,
+            lr=0.05,
+            batch_size=33,
+            criterion__reduction='sum',
```

Comment:
Does the reduction need to be sum here? In practice, I think most use the default 'mean' reduction.

Reply:
I just chose something to change the parameter of each component, in this case the criterion, to a different value, to make sure that even if re-initialized, we're still fine. What exactly that value is doesn't matter.